husky-fhir/husky-fhir-emed-ch/husky-fhir-emed-ch-epr-narrative/src/main/java/org/projecthusky/fhir/emed/ch/epr/narrative/pdf/HtmlToPdfAConverter.java

```diff
@@ -22,28 +23,17 @@
 public interface HtmlToPdfAConverter {
 
     /**
-     * Converts the HTML content to its PDF/A representation.
+     * Converts the HTML content to its PDF/A representation, with a custom template.
      *
-     * @param subject     The document subject. It's inserted in a {@code <meta name="subject">} tag.
-     * @param author      The document author. It's inserted in a {@code <meta name="author">} tag.
-     * @param description The document description. It's inserted in a {@code <meta name="description">} tag.
-     * @param title       The document title.
-     * @param lang        The document language, as an ISO code.
-     * @param bookmarks   A list of bookmarks to interesting parts of the document. The keys are labels that are shown
-     *                    to the user, the values are HTML ID references (starting with '#').
-     * @param cssContent  The CSS content to inject.
-     * @param bodyContent The HTML content to convert. It will be inserted in a {@code <body>} tag.
+     * @param lang             The document language, as an ISO code.
+     * @param contextVariables The context variables to be added to the template processing. Note that {@code "lang"} is
+     *                         automatically added with the code value of the received {@code lang} parameter.
+     * @param templateContent The custom Thymeleaf content.
      * @return The content of the generated PDF/A.
-     * @throws Exception it may throw any exception.
      */
-    @SuppressWarnings("java:S112")
     // Allow any exception to be thrown
-    byte[] convert(final String subject,
-                   final String author,
-                   final String description,
-                   final String title,
-                   final NarrativeLanguage lang,
-                   final Map<String, String> bookmarks,
-                   final String cssContent,
-                   final String bodyContent) throws Exception;
+    byte[] convert(final NarrativeLanguage lang,
+                   final Map<String, Object> contextVariables,
```

Comment:
Not sure about this change, the variables subject, author, description, title and bookmarks are needed for a PDF-A document. We should make sure they are provided to the generator, otherwise it might produce an invalid PDF-A document.

Reply:
I see your point, I will change it so that those variables are explicitly passed to the method.

Reply:
Coming back to this. I was about to address this issue but I had some doubts about it. The issue I have with providing those variables explicitly is that in any case we allow for a completely custom template to be provided. This means that in the template those values might be even hardcoded and the variables that we would be passing here would be useless. Wouldn't this normally be more of a documentation issue? Would it be enough to add this information on the javadoc and if needed in any extra documentation regarding the pdf converter?

Reply:
That's a valid point. Okay, let's use the javadoc for information, and let the implementors decide which variables will be needed in their templates.